geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java

```diff
@@ -25,76 +27,63 @@
 /**
  * This class acts as a monitor and listen for VM stats update on behalf of MemberMBean.
  *
- *
  */
 public class VMStatsMonitor extends MBeanStatsMonitor {
-
-  private static final int VALUE_NOT_AVAILABLE = -1;
-
-  private volatile float cpuUsage = 0;
-
-  private static String processCPUTimeAttr = "ProcessCpuTime";
-
+  static final float VALUE_NOT_AVAILABLE = -1;
+  private static final String PROCESS_CPU_TIME_ATTRIBUTE = "ProcessCpuTime";
   private long lastSystemTime = 0;
-
   private long lastProcessCpuTime = 0;
+  private final boolean processCPUTimeAvailable;
+  private final AtomicInteger cpuUsageBits = new AtomicInteger(Float.floatToIntBits(0f));
 
-  private boolean processCPUTimeAvailable;
+  public float getCpuUsage() {
+    return Float.intBitsToFloat(cpuUsageBits.get());
+  }
 
   public VMStatsMonitor(String name) {
     super(name);
-    processCPUTimeAvailable = MBeanJMXAdapter.isAttributeAvailable(processCPUTimeAttr,
+    processCPUTimeAvailable = MBeanJMXAdapter.isAttributeAvailable(PROCESS_CPU_TIME_ATTRIBUTE,
         ManagementFactory.OPERATING_SYSTEM_MXBEAN_NAME);
     if (!processCPUTimeAvailable) {
-      cpuUsage = VALUE_NOT_AVAILABLE;
+      cpuUsageBits.set(Float.floatToIntBits(VALUE_NOT_AVAILABLE));
     }
-
   }
 
-
-  @Override
-  public void handleNotification(StatisticsNotification notification) {
-
-    for (StatisticId statId : notification) {
-      StatisticDescriptor descriptor = statId.getStatisticDescriptor();
-      String name = descriptor.getName();
-      Number value;
-      try {
-        value = notification.getValue(statId);
-      } catch (StatisticNotFoundException e) {
-        value = 0;
-      }
-      log(name, value);
-      statsMap.put(name, value);
-    }
-    refreshStats();
+  long currentTimeMillis() {
+    return Instant.now().toEpochMilli();
```

Comment:
I'm not sure this is any better. It seems to me like you might still be converting to wall clock time.
Would System.nanoTime() converted to millis be better?